### PR TITLE
Add multiplier parameter to transaction_payment function

### PR DESCRIPTION
**Pull request by svyatonik**
This change is required for all Polkadot-like chains

### CODE REVIEW

File: bin/runtime-common/src/messages.rs [hunk 1]
```diff
@@ -160,9 +166,8 @@ pub fn transaction_payment_without_multiplier<Balance: AtLeast32BitUnsigned>(
 	let len_fee = per_byte_fee.saturating_mul(Balance::from(transaction.size));
 
 	// the adjustable part of the fee
-	//
-	// here we assume that the fee multiplier is zero, so this part is also always zero
-	let adjusted_weight_fee = Balance::zero();
+	let unadjusted_weight_fee = weight_to_fee(transaction.dispatch_weight);
+	let adjusted_weight_fee = multiplier.saturating_mul_int(unadjusted_weight_fee);
```
Comment (tomusdrw) [inline bin/runtime-common/src/messages.rs:170]: Can you add a test for this function or add one that set's non-zero multiplier?

File: bin/millau/runtime/src/rialto_messages.rs [hunk 2]
```diff
@@ -146,9 +146,10 @@ impl messages::ThisChainWithMessageLanes for Millau {
 
 	fn transaction_payment(transaction: MessageLaneTransaction<Weight>) -> bp_millau::Balance {
 		// in our testnets, both per-byte fee and weight-to-fee are 1:1
-		messages::transaction_payment_without_multiplier(
+		messages::transaction_payment(
 			bp_millau::BlockWeights::get().get(DispatchClass::Normal).base_extrinsic,
 			1,
+			FixedU128::zero(),
```
Comment (tomusdrw) [inline bin/millau/runtime/src/rialto_messages.rs:152]: Is there a way how we can test non-zero values on rialto<>millau? What does the multiplier depend on? Is it block utilisation?

Comment (svyatonik) [reply bin/millau/runtime/src/rialto_messages.rs:152]: > Is there a way how we can test non-zero values on rialto<>millau?

No

> What does the multiplier depend on? Is it block utilisation?

This is defined by `pallet_transaction_payment::FeeMultiplierUpdate`. In P+K if block has weight > 25% of its maximal weight, then this multiplier grows. If less than 25%, then it drops (to some minimal value).

File: bin/runtime-common/src/messages.rs [hunk 3]
```diff
@@ -1401,4 +1406,38 @@ mod tests {
...
+				100,
+				10,
+				FixedU128::zero(),
+				|weight| weight,
+				MessageLaneTransaction {
+					size: 50,
+					dispatch_weight: 777
+				},
+			),
+			100 + 50 * 10,
```
Comment (HCastano) [inline bin/runtime-common/src/messages.rs:1423]: nit: Having something like `base_weight + size * per_byte_fee` here would be nice

**Review by HCastano — APPROVED** [review]

File: bin/runtime-common/src/messages.rs (hunk 3 above)
Comment (svyatonik) [reply bin/runtime-common/src/messages.rs:1423]: Then it'll be duplicate of the `transaction_payment` itself :) What it does is simply gives names to these numbers
